File: 27 - Connect with JS/a. CSharp calling JS - iv. JSImport - With Return/wasm/App.razor.cs
// European Union Public License version 1.2
// Copyright © 2023 Rick Beerendonk

using System.Runtime.InteropServices.JavaScript;

namespace CSharpCallingJS_WithReturn;

public partial class App
{
    [JSImport("jsfunc", "App")]
    internal static partial double JSFunc(string input);
}

File: 27 - Connect with JS/a. CSharp calling JS - iv. JSImport - With Return/wasm/App.razor.js
// European Union Public License version 1.2
// Copyright © 2023 Rick Beerendonk

// Function to call from Blazor code.
export function jsfunc(input) {
  console.log("JS received:", input);
  const output = Math.random();
  console.log("JS returns:", output);
  return output;
}
